packages/auth/amplify_auth_cognito_dart/lib/src/state/state/credential_store_state.dart

```diff
@@ -181,7 +183,11 @@ class CredentialStoreFailure extends CredentialStoreState with ErrorState {
 /// {@template amplify_auth_cognito_dart.credential_store_state.credential_store_data}
 /// Data stored by the Credential Store.
 /// {@endtemplate}
-class CredentialStoreData with AWSEquatable<CredentialStoreData> {
+@zAmplifySerializable
+class CredentialStoreData
```

Comment:
is there a security risk on serializing credential data?

Reply:
None that I can think of. The main risk around credentials is accidentally leaking them via logs - we are protected against that here by not overriding `toString`, so by default, printing an instance of this class will show (`Instance of 'CredentialStoreData'`)

Reply:
you are right, it does not add risk on accidental leak via prinitng an instance. do you see risk of leaking or malicious actions on instance.toJson() as we are making it serializable?

Reply:
No more risk than exposing the class to begin with, I think. Since you have to explicitly call `toJson` as the developer, just like you would explicitly have to retrieve instance variables if you wanted to print them or send them over the wire somewhere.